plasmapy/simulation/particle_tracker.py

```diff
@@ -252,6 +252,10 @@ def post_push_hook(self, force_save=False) -> None:
         if self.save_now or force_save:
             self.save()
 
+    def post_run_hook(self) -> None:
+        """Function called after the simulation has finished running."""
+        return
```

Comment:
This is the best way I could come up with to create an "optional" abstract method (i.e. a method that is guaranteed to be implemented but whose implementation doesn't necessarily have to come from the child class)